pkg/services/dns/dns.go

```diff
@@ -141,7 +157,7 @@ func (s *Server) Serve() error {
 
 func (s *Server) ServeTCP() error {
 	mux := dns.NewServeMux()
-	mux.HandleFunc(".", s.handler.handle)
```

Comment:
This makes `handle` only serve UDP?

Reply:
Yeah unfortunately since the size passed into the truncate needs to be different between UDP and TCP, unless there is a way to determine if it's TCP or UDP in the dns request but I couldn't find it

Reply:
Just implemented it with the size option.. I went with the logic specified in the comments on the truncate method:
https://github.com/miekg/dns/blob/730c265d18c3f51b447f6a4c61fe3cde336585d8/msg_truncate.go#L25

Reply:
I was playing around with different buffer sizes with dig. You can see when dig see's truncated UDP responses it will auto -retry over TCP. But the truncation seems to be working well within the bufsize passed
 `dig +bufsize=1000 aerserv-bc-us-east.bidswitch.net`
  `dig +tcp +bufsize=512 aerserv-bc-us-east.bidswitch.net`